src/cheri/riscv-unpriv-integration.adoc

```diff
@@ -109,280 +304,352 @@ Additionally, CSRs that hold pointers are extended to capability, but since no C
-(draw-box "tag" {:span 1 :borders {}}) (draw-box "metadata" {:span 32 :borders {}}) (draw-box "address" {:span 32 :borders {}})
+(draw-box "V" {:span 1 :borders {}}) (draw-box "Metadata" {:span 32 :borders {}}) (draw-box "Address" {:span 32 :borders {}})
 ----
 
-* Tags in registers, caches, and memory:
+==== Extended general purpose integer registers
+The <<gprs,XLEN-wide integer registers>> (e.g. `sp`, `a0`) are all extended with another XLEN bits of capability metadata plus a validity tag bit.
+When referring to the extended register, the `x` prefix is replaced with a `c`: i.e. the capability extended version of `x0` becomes `c0`.
+The existing integer register names refer to the address part of the capability.
+For the ABI register names the `c` prefix is added, i.e. `csp`, `ca0`.
+The zero register is extended with zero metadata and a cleared validity tag: this is called the <<null-cap>> capability.
+
+NOTE: Instruction encodings refer to capability registers when they use `cs1`, `cd`, etc. instead of `rs1`, `rd` (which reads/writes the XLEN-bit subset of the register).
 
-** Every register has a one-bit tag, indicating whether the capability in the register is valid to be dereferenced.
-This tag is cleared whenever an invalid capability operation is performed. Such invalid operations include e.g. writing only the integer portion of the register or attempting to increase bounds or permissions.
+==== Extended CSRs
+All registers that store addresses are extended to contain capabilities.
+In the base {cheri_base_ext_name} ISA, the only such register is the program counter `pc` that is extended to the Program Counter Capability (<<pcc>>).
 
-** The tags are also tracked through the memory subsystem: every aligned CLEN-bits wide region has a non-addressable one-bit tag, which the hardware manages atomically with the data.
-The tag is cleared if the memory region is ever written using an operation other than a store of a full capability register.
-Any caches must preserve this abstraction.
+NOTE: All CSRs that hold pointers are extended to CLEN bits, but since no CSRs are defined in the RV32I/RV64I base instruction set, any such change is described in the corresponding chapters.
 
-==== Instructions using capability registers
+[#pcc,reftext="pcc"]
+===== The Program Counter Capability (`pcc`)
 
-{cheri_base_ext_name} adds new instructions to interact with the extended registers.
-These added instructions can be split into the following categories:
+The `pc` is extended to be the Program Counter Capability.
+Extending the `pc` allows the range of branches, jumps and linear execution for currently executing code to be restricted.
+
+The <<pcc>> address field is the `pc` in the base RISC-V ISA so that the
+hardware automatically updates it as instructions are executed. The <<pcc>>'s
+metadata and validity tag are reset to the <<infinite-cap>> capability values,
```

Comment:
Again this is not exactly true for all CHERI implementations. Could we be more vague here and say that the PC should be reset to a capability that can execute from anywhere in memory?

Reply:
Updated.